### ISSUE TITLE
Map commit SHA to tag

### ISSUE THREAD

**Issue by estahn** [getsentry/sentry#51860]
### Problem Statement

We currently deploy our application before tagging it. This is mainly legacy to ensure the deploy goes through and is valid. We notify sentry as part of the deploy and hence all releases will receive a SHA instead of a semantic version.

### Solution Brainstorm

Allow to map a tag to a commit via e.g. sentry-cli or GitHub Action without creating a new release.

### Product Area

Unknown

**Comment by estahn** [comment]
@scttcper Example:

1. github-action pushes SHA `66fc1ec7346f02a57a6a9e1a5c3b4b0e04aae473` as a release via sentry github action
2. We tag `66fc1ec7346f02a57a6a9e1a5c3b4b0e04aae473` with `v1.2.3`

How do we let sentry know that release `66fc1ec7346f02a57a6a9e1a5c3b4b0e04aae473` should be displayed as `v1.2.3`?

**Comment by scttcper** [comment]
I see. Right now, you'd want to tell the github action and the sdk about v1.2.3 and not the commit sha since you can't rename it later.

I can add this to our backlog

**Comment by shuneihayakawa** [comment]
Any updates on this feature? This will be really helpful for our team too.

Currently we are building containerized Next.js application. When commit is pushed to branch, the container image is built and during the build, we uploads source maps and create Sentry release. After some testing, we will create release on Github, which triggers the deploy to production environment. 
In this case, we can't use version  number as the name of Sentry release, since the build step and deploy step is separated and we can't know the version number at build time.
I think separating the build and deployment phase is quite common in containerized applications.
I hope Sentry team will work on this feature !

**Comment by snigdhas** [comment]
I don't believe this has been added to our roadmap yet - I've included it in our backlog for prioritization.
